cachito/workers/scm.py

```diff
@@ -158,9 +158,13 @@ def clone_and_archive(self, gitsubmodule=False):
                     self.url,
                     clone_path,
                     no_checkout=True,
+                    # Clone with depth 1 to save resources on large repos
+                    depth=1,
                     # Don't allow git to prompt for a username if we don't have access
                     env={"GIT_TERMINAL_PROMPT": "0"},
                 )
+
+                repo.remote().fetch(refspec=self.ref, depth=1)
```

Comment:
Why do we need to have this line twice?

Reply:
Because the shallow clone always defaults to the lastest master commit, so we need to fetch the commit we're actually building.

Reply:
Aaah, I just forgot there are some lines between 170 and 194. My bad!